onadata/apps/main/forms.py

```diff
@@ -324,6 +331,13 @@ def publish(self, user, id_string=None, created_by=None):
             else:
                 project = self._project
 
+            if 'xml_file' in self.cleaned_data and\
```

Comment:
The key might exist but have an empty string, thus the need for the second check.